script/vsts/platforms/templates/preparation.yml

```diff
@@ -3,8 +3,11 @@ steps:
   # Linux Specific
   - script: |
       sudo apt-get update
-      sudo apt-get install -y wget software-properties-common
-      sudo apt-get install -y build-essential ca-certificates xvfb fakeroot git libsecret-1-dev rpm libx11-dev libxkbfile-dev xz-utils xorriso zsync libxss1 libgtk-3-0 libasound2 libicu-dev
+      sudo apt-get install -y build-essential ca-certificates xvfb fakeroot git libsecret-1-dev libx11-dev libxkbfile-dev xz-utils xorriso zsync libxss1 libgtk-3-0 libasound2 libicu-dev software-properties-common wget
+      # rpm needs to be at least 4.14, in xenial it is too old
+      sudo apt-add-repository -y 'deb http://azure.archive.ubuntu.com/ubuntu/ bionic main universe'
```

Comment:
Can you add the Bionic security updates repo as well?

`deb http://security.ubuntu.com/ubuntu bionic-security main`

There are a couple of the packages with security patched versions that only landed in that repository.

_(I copied that repo URL from here: https://packages.ubuntu.com/bionic/amd64/libdw1/download, https://packages.ubuntu.com/bionic/amd64/libzstd1/download, https://packages.ubuntu.com/bionic/amd64/libelf1/download. The other packages' latest versions are in the non-security repo. It does not look like Azure mirrors the security repos, which I suppose means no delay in receiving the patches.)_

Reply:
Doesn't azure have a copy at http://azure.archive.ubuntu.com/ubuntu/dists/bionic-security/ ?

Reply:
I put in now this Azure security mirror, and it served the same updated version as `security.ubuntu.com` so I think we are good now :-)

Reply:
Even better, when checking the Xenial repositories that are already being used in this image, I noticed that we can use "Extended Security Maintenance' instead, to be more future-proof.

Reply:
Scratch that last bit, Azure is more up to date than ESM :-/

Reply:
> Doesn't azure have a copy at http://azure.archive.ubuntu.com/ubuntu/dists/bionic-security/ ?

It appears you are right. I wonder why they do not use their own mirror for `xenial-security`. According to the CI logs, they only fetch from the `ubuntu.com` copy of `xenial-security`. :shrug: